chef/cookbooks/barclamp/spec/libraries/barclamp_library_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "spec_helper"
+
+require_relative "../../libraries/barclamp_library"
+
+describe BarclampLibrary::Barclamp::Inventory::Disk do
+
+  before(:each) do
+    @chef_run = ::ChefSpec::Runner.new
+    @node = @chef_run.node
+    @node.default[:block_device] = {
+      dm0: { removable: "0" },
+      xvd1: { removable: "0" },
+      xvd2: { removable: "1" }
+    }
+  end
+
+  it "unclaimed method returns the proper number of unclaimed devices" do
+    a = BarclampLibrary::Barclamp::Inventory::Disk
+    expect(a).to receive(:`).exactly(5).times.and_return(`exit 1`)
```

Comment:
yeah this is wonderful.

So this is mocking the calls that are done to `%x{lsblk #{d.name.gsub(/!/, "/")} --noheadings --output MOUNTPOINT | grep -q -v ^$}` in the `unclaimed` method, but as we check the `$?.exitstatus == 0` I could not find any other proper way of mocking that correctly in a simple way, so I just mock the system call and return an actual `exit 1` and save in complexity :)

> But that's just the empty string

its actually not an empty string due to the special `$?` :O